src/main/scala/io/iohk/ethereum/db/dataSource/LevelDBDataSource.scala

```diff
@@ -81,7 +73,21 @@ class LevelDBDataSource(
     try {
       close()
     } finally {
-      Iq80DBFactory.factory.destroy(new File(levelDbConfig.path), null) // Options are not being used ¯\_(ツ)_/¯
+      import levelDbConfig._
+
+      val options = new Options()
+        .createIfMissing(createIfMissing)
+        .paranoidChecks(paranoidChecks) // raise an error as soon as it detects an internal corruption
+        .verifyChecksums(verifyChecksums) // force checksum verification of all data that is read from the file system on behalf of a particular read
+
+
+      val factory = if (native) {
+        org.fusesource.leveldbjni.JniDBFactory.factory
```

Comment:
If we would like to support LevelDb jni that way we woulnd need similar `ReadWriteLock` as in rocks db to avoid corruption on close or open. This is one reasons to maybe consider supporting only rocks db and java leveldb as fallback.